pkg/options/auth.go

```diff
@@ -48,57 +48,40 @@ const (
 //	      hostname: example.com
 //	      port: 80
 type AuthOptions struct {
-	// "basic", "cloudentity" "oauth2".
-	// REQUIRED.
-	Scheme string `json:"scheme,omitempty" yaml:"scheme,omitempty"`
-	// OPTIONAL. TODO(MBana): Move to `AuthUpstream`.
-	PathPrefix *string `json:"path_prefix,omitempty" yaml:"path_prefix,omitempty"`
-	// REQUIRED, if `scheme == basic`. Mutually exclusive with `OAuth2`.
-	AuthUpstream *AuthUpstream `json:"auth-upstream,omitempty" yaml:"auth-upstream,omitempty"`
-	// REQUIRED, if `scheme == oauth2`. Mutually exclusive with `AuthUpstream`.
+	// OPTIONAL
 	OAuth2 *OAuth2 `json:"oauth2,omitempty" yaml:"oauth2,omitempty"`
+	// OPTIONAL
+	Custom *Custom `json:"custom,omitempty" yaml:"custom,omitempty"`
+	// OPTIONAL
+	Cloudentity *Cloudentity `json:"cloudentity,omitempty" yaml:"cloudentity,omitempty"`
 }
 
 func (o AuthOptions) String() string {
 	return ToCompactJSON(o)
 }
 
 func (o AuthOptions) Validate() error {
-	err := validation.ValidateStruct(&o,
-		validation.Field(&o.Scheme, validation.Required, validation.In(SchemeBasic, SchemeCloudEntity, SchemeOAuth2)),
-	)
-	if err != nil {
-		return err
+	if o.OAuth2 == nil && o.Custom == nil && o.Cloudentity == nil {
+		return fmt.Errorf("auth must have one of the following defined OAuth2, Custom or Cloudentity")
 	}
 
-	// `basic` or `cloudentity`
-	if SchemeBasic == o.Scheme || SchemeCloudEntity == o.Scheme {
-		return validation.ValidateStruct(&o,
-			validation.Field(&o.AuthUpstream, validation.Required),
-		)
+	if o.OAuth2 != nil && o.Custom != nil {
+		return fmt.Errorf("custom auth and OAuth2 cannot be enabled at the same time")
```

Comment:
I think we need a generic check: 
if: more than 1 auth mechanism is defined 
return: Only 1 authentication mechanism allowed

or something like that? 🤔

Reply:
That's already in place